File: Escola/Cursos.Repository/Services/InscricaoService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  InscricaoService
      /// </summary>
      public partial class InscricaoService : BaseServices, IInscricaoService
      {

		public InscricaoService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_INSCRICAO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_INSCRICAO> entity)
        {
            UnitOfWork.InscricaoRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_INSCRICAO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_INSCRICAO entity)
        {	
            UnitOfWork.InscricaoRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_INSCRICAO
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_INSCRICAO entity)
        {
            UnitOfWork.InscricaoRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_INSCRICAO
        /// </summary>
        /// <returns>IEnumerable EV_INSCRICAO</returns>
        public IEnumerable<EV_INSCRICAO> GetAll()
        {
			return UnitOfWork.InscricaoRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_INSCRICAO
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_INSCRICAO entity)
        {
            UnitOfWork.InscricaoRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_INSCRICAO By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_INSCRICAO</returns>
		public EV_INSCRICAO GetById(int id)
        {
			return UnitOfWork.InscricaoRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_INSCRICAO> FindBy(Expression<Func<EV_INSCRICAO, bool>> predicate)
        {
            return UnitOfWork.InscricaoRepository.FindBy(predicate);
        }
      }
}

File: Escola/Cursos.Api/scripts/service/inscricaoService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("InscricaoService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Inscricao/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Inscricao/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Inscricao, callback) {
            $http.post(serviceBase + "Inscricao/post", Inscricao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Inscricao, callback) {
            $http.post(serviceBase + "Inscricao/put", Inscricao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Inscricao, callback) {
            $http.post(serviceBase + "Inscricao/delete", Inscricao)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

